src/sql/parts/notebook/cellViews/codeActions.ts

```diff
@@ -10,97 +12,176 @@ import { CellType } from 'sql/parts/notebook/models/contracts';
 import { NotebookModel } from 'sql/parts/notebook/models/notebookModel';
 import { getErrorMessage } from 'sql/parts/notebook/notebookUtils';
 import { INotificationService, Severity } from 'vs/platform/notification/common/notification';
-import { NOTFOUND } from 'dns';
-import { NsfwWatcherService } from 'vs/workbench/services/files/node/watcher/nsfw/nsfwWatcherService';
+import { ICellModel, FutureInternal } from 'sql/parts/notebook/models/modelInterfaces';
+import { ToggleableAction } from 'sql/parts/notebook/notebookActions';
 
 let notebookMoreActionMsg = localize('notebook.failed', "Please select active cell and try again");
-export class RunCellAction extends Action {
-	public static ID = 'jobaction.notebookRunCell';
-	public static LABEL = 'Run cell';
 
-	constructor(
-	) {
-		super(RunCellAction.ID, '', 'toolbarIconRun');
-		this.tooltip = localize('runCell', 'Run cell');
+function hasModelAndCell(context: CellContext, notificationService: INotificationService): boolean  {
+	if (!context || !context.model) {
+		return false;
```

Comment:
Just checking - Don't we need to notify with error message if model is undefined similar to Cell below?

Reply:
Good question. This was Yurong's implementation that I just ported to a common location. In the long term if context is missing, we should query for active notebook / cell. In the short term I think this would indicate the model hasn't loaded yet so buttons should just no-op.